bin/propolis-cli/src/main.rs

```diff
@@ -288,22 +298,39 @@ impl VmConfig {
             return parse_json_file(path);
         }
 
-        let from_toml = &self
+        let parsed_toml = self
             .config_toml
             .as_ref()
             .map(propolis_config_toml::parse)
-            .transpose()?
-            .as_ref()
-            .map(SpecConfig::try_from)
             .transpose()?;
 
+        let from_toml =
+            parsed_toml.as_ref().map(SpecConfig::try_from).transpose()?;
+
         let enable_pcie =
             from_toml.as_ref().map(|cfg| cfg.enable_pcie).unwrap_or(false);
 
+        let cpuid_profile = parsed_toml
+            .as_ref()
+            .and_then(|cfg| {
+                self.cpuid_profile.as_ref().map(|profile| {
+                    cfg.cpuid_profiles
+                        .get(profile)
+                        .ok_or_else(|| {
+                            anyhow!("CPUID profile not defined: {}", profile)
```

Comment:
turbo nit
```suggestion
                            anyhow!("CPUID profile not defined: {profile}",)
```
also, maybe it would be nice to have this say "not defined in path/to/toml.toml"?

Reply:
i'll internalize in-line format variables one day.. i was less sure about including the toml path at first because i'd like to check the name against a the kinds of profiles we've got in Nexus (after hoisting `cpu_platform.rs` out of Omicron, probably), but we can change the way this error is printed when that time comes